nixos/modules/services/web-servers/caddy.nix

```diff
@@ -6,90 +6,101 @@ let
   cfg = config.services.caddy;
   configFile = pkgs.writeText "Caddyfile" cfg.config;
 in
-{
-  options.services.caddy = {
-    enable = mkEnableOption "Caddy web server";
+  {
+    options.services.caddy = {
+      enable = mkEnableOption "Caddy web server";
 
-    config = mkOption {
-      description = "Verbatim Caddyfile to use";
-    };
+      config = mkOption {
+        default = "";
+        example = ''
+          example.com {
+          gzip
+          minify
+          log syslog
 
-    ca = mkOption {
-      default = "https://acme-v01.api.letsencrypt.org/directory";
-      example = "https://acme-staging.api.letsencrypt.org/directory";
-      type = types.string;
-      description = "Certificate authority ACME server. The default (Let's Encrypt production server) should be fine for most people.";
-    };
+          root /srv/http
+          }
+        '';
+        type = types.string
```

Comment:
Oops! Sorry about that. And cool, I wasn't aware of `types.lines`. Changes made and squashed with the original commit.